### ISSUE TITLE
Looks for path information with Unix patterns

### ISSUE THREAD

**Issue by gregmacfarlane** [pbsag/swimctr#1]
When determining if a parameter is a value or a path, the code looks to see if the string begins with `/`.

```r
      # But otherwise we will attempt to read the file as CSV format. However,
      # at this point we have no idea whether the filename has path included or
      # not. So we'll see if the first character is a slash, and if so, assume
      # that it does. If not we'll preface the filename with root.dir.
      first_char <- substr(params$value[i], 1, 1)
      if (first_char != "/") {
        if (nchar(RTP[["root.dir"]])<1) {
          error_message <- paste("Somehow root.dir is not defined at seq=", 
            params$seq[i])
          stop(error_message)
        }
        filename <- file.path(RTP[["root.dir"]], params$value[i])
      } else {
        filename <- params$value[i]
      }
```

It seems as though this will not work on Windows systems where paths typically start with 'X:\'

**Comment by gregmacfarlane** [comment]
For instance, on Windows when I give `& C:/model/outputs/file.csv` as the path, it cannot read `C:/model//C:/model/outputs/file.csv` because it pre-appends the `root.dir` attributes.

**Comment by rickdonnelly** [comment]
I had formerly checked to see if the parameter token included a leading folder name if it was a filename, which I could have used dirname() to check for. It normally returns "." if you just specify a filename with no leading directory (i.e., is in the same folder R is started from). But dirname() returned garbage in earlier version of Windows 10. I developed and tested my code on Mac, but then checked that it would run under Windows 10, as I don't have access to machine running Windows 7 or 8.1 any longer (or desire to). Since I never include the logical drive letter even when using Windows it never occurred to me to check for that. I just verified that dirname() now works correctly under Windows 10 (build 1067) and R 3.3.2, but then it occurred to me that I don't need to parse the filename in the first place. I can just see if it exists, and if so, read it. If not, try appending the filename to root.dir and see if it exists. If not, or if root.dir (a required runtime parameter) is not set, then crash. I've made that update to get_runtime_parameters(), and verified that new code works on both Windows 10 and Windows Server 2016 (and on macOS Sierra, of course).
